tests/tests_integration/test_api/test_geospatial.py

```diff
@@ -299,6 +299,37 @@ def test_search_single_feature(self, cognite_client, test_feature_type, test_fea
         )
         assert len(res) == 0
 
+    def test_search_feature_dimensionality_mismatch(self, cognite_client, test_feature_type, test_feature):
+        polygon_z = "POLYGONZ((2.276 48.858 3,2.278 48.859 3,2.2759 48.859 3,2.276 48.858 3))"
+        polygon = "POLYGON((2.276 48.858,2.278 48.859,2.275 48.859,2.276 48.858))"
+        res = cognite_client.geospatial.search_features(
+            feature_type_external_id=test_feature_type.external_id,
+            filter={"stWithin": {"property": "position", "value": {"wkt": polygon}}},
+            limit=10,
+        )
+        assert res[0].external_id == test_feature.external_id
+
+        try:
+            res = cognite_client.geospatial.search_features(
+                feature_type_external_id=test_feature_type.external_id,
+                filter={"stWithin": {"property": "position", "value": {"wkt": polygon_z}}},
+                limit=10,
+            )
+            pytest.fail("searching features with wrong dimensionality did not raise an exception")
+        except CogniteAPIError:
+            pass
```

Comment:
Generally, I favor this pattern:
```py
with pytest.raises(CogniteAPIError):
   ...
```
That way you assert that the error is always raised. You can also use the `match` parameter to verify some part of the actual error message raised, if useful.

Reply:
(Then you can remove the `pytest.fail`)